### PR TITLE
adding UA_Client_NamespaceGetIndex

### LINKED ISSUE #405: UA_NamespaceGetIndex for Clients
For implementing the PLCOPEN Client FBs we need the function `UA_NamespaceGetIndex` probably with the prototype
`UA_UINT UA_NamespaceGetIndex(UA_Client *client, UA_String *NamespaceUri);`

### PR DESCRIPTION
**Pull request by HolgerJeromin**
fixes #405

### CODE REVIEW

**Comment by HolgerJeromin** [comment]
Should we ifdef this code with high level abstraction api?

File: src/client/ua_client.c [hunk 1]
```diff
@@ -629,6 +629,44 @@ UA_BrowseResponse UA_Client_browse(UA_Client *client, UA_BrowseRequest *request)
...
+	ReadRequest.nodesToRead[0].nodeId = UA_NODEID_NUMERIC(0, UA_NS0ID_SERVER_NAMESPACEARRAY);
+
+	ReadResponse = UA_Client_read(client, &ReadRequest);
+	if(ReadResponse.responseHeader.serviceResult == UA_STATUSCODE_GOOD){
+		retval = UA_STATUSCODE_BADTYPEMISMATCH;
+		if(ReadResponse.resultsSize > 0 && ReadResponse.results[0].hasValue){
+			if(!UA_Variant_isScalar(&ReadResponse.results[0].value)){
+				if(ReadResponse.results[0].value.type == &UA_TYPES[UA_TYPES_STRING]){
+					retval = UA_STATUSCODE_BADNOTFOUND;
+					for(UA_Int32 iterator = 0; iterator < ReadResponse.results[0].value.arrayLength;iterator++){
```
Comment (Stasik0) [inline src/client/ua_client.c]: y not UInt16 as variable? No need to cast in this case

**Comment by Stasik0** [comment]
@HolgerJeromin the build is errored, please resync with master

File: src/client/ua_client.c (hunk 1 above)
Comment (HolgerJeromin) [inline src/client/ua_client.c]: arrayLength is int32

Comment (Stasik0) [inline src/client/ua_client.c]: Yea, but an upcast is always safe, whie downcast is not.

**Comment by HolgerJeromin** [comment]
I am too lazy (incompetent?) to rebase, but the travis/pr build does the merging for me :-)

**Comment by Stasik0** [comment]
squash into one? hi-level ifdefs?

**Comment by HolgerJeromin** [comment]
Is there a hi level define? a quick search in master / hl branch was not successful.

BTW i mailed github to support squashing via webinterface :-)

**Comment by HolgerJeromin** [comment]
^- very proud of his first commit squashing :-)

**Comment by HolgerJeromin** [comment]
@jpfr @ichrispa 
Is there or is there no `define` available for high level API functions?

**Comment by Stasik0** [comment]
@jpft @FlorianPalm  @HolgerJeromin I re-shaped the code into a not-nested way by using a goto, everyone okay with that?

**Comment by HolgerJeromin** [comment]
Thank you both

**Pull request merged**